Buildings/Resources/src/ThermalZones/Detailed/Validation/BESTEST/simulateAndPlot.py

```diff
@@ -0,0 +1,988 @@
+#!/usr/bin/env python
+
+import matplotlib
+matplotlib.use('Agg')
+
+import matplotlib.pyplot as plt
+
+import io
+import json
+import os
+import sys
+import shutil
+
+# check if it just implements post-process
+POST_PROCESS_ONLY = False
+# check if delete the simulation result files
+CLEAN_MAT = True
+# check if delete the temporary working folders
+DelTemDir = True
+
+CWD = os.getcwd()
+FROM_GIT_HUB = True
+# Modelica Buildings Library working branch
+MBL_BRANCH = 'master'
+# BuildingSpy working branch. 
+# The working branch makes the communication points ('ncp') of jmodelica or optimica to be 8761,
+# so that the regression test will generate high resolution results.
+BP_BRANCH = 'issue335_high_ncp'
+# simulator, JModelica and optimica are supported
+TOOL = 'optimica'
+
+# standard data file
+ASHRAE_DATA = './ASHRAE140_data.dat'
+PACKAGES = ['Buildings.ThermalZones.Detailed.Validation.BESTEST.Cases6xx', \
+            'Buildings.ThermalZones.Detailed.Validation.BESTEST.Cases9xx']
+CASES = ['Case600', 'Case600FF', 'Case610', 'Case620', 'Case630', 'Case640', 'Case650', 'Case650FF', \
+         'Case900', 'Case900FF', 'Case920', 'Case940', 'Case950', 'Case950FF', 'Case960']
+
+plt.rc('axes', labelsize=9)
+plt.rc('xtick', labelsize=9)
+plt.rc('ytick', labelsize=9)
+plt.rc('legend', fontsize=9)
+plt.rcParams['axes.facecolor']='whitesmoke'
+
+def save_plot(figure, file_name):
+    """ Save the figure to a pdf and png file in the directory `img`
+    """
+
+    out_dir = "../../../../../Images/ThermalZones/Detailed/Validation/BESTEST"
+    if not os.path.exists(out_dir):
+        os.makedirs(out_dir)
+    figure.savefig(os.path.join(out_dir, '{}.pdf'.format(file_name)))
+    figure.savefig(os.path.join(out_dir, '{}.png'.format(file_name)))
+    plt.clf()
+
+def configure_axes(axes):
+    """ Configure the axis style
+    """
+    axes.spines['right'].set_linewidth(0.25)
+    axes.spines['right'].set_color('lightgrey')
+    axes.spines['top'].set_visible(False)
+    axes.spines['left'].set_linewidth(0.25)
+    axes.spines['left'].set_color('lightgrey')
+    axes.spines['bottom'].set_visible(False)
+    axes.grid(color='lightgrey', linewidth=0.25)
+    return
+
+def create_library_directory(repoNam):
+    ''' Create directory to be as temporary directory to save buildings library.
+    '''
+    import tempfile
+    import getpass
+    prefixName = 'tmp-MBL-' if repoNam == 'MBL' else 'tmp-BP-'
+    worDir = tempfile.mkdtemp( prefix=prefixName + getpass.getuser() )
+    print("Created directory {}".format(worDir))
+    return worDir
+
+def checkout_repository(working_directory, repoNam, from_git_hub, branch):
```

Comment:
This function should be changed: You should only check out the special version of BuildingsPy, and make sure it is at the beginning of the `PYTHONPATH` so that not the already installed `BuildingPy` is used.
The Buildings library can just be copied to a temporary folder, no need to check it out.

Reply:
Updated. Now it will copy buildings library to temporary folder, no check out. For buildingspy, it clones the repository to temporary folder and checks out to target branch. The regression will use the `regresstiontest.py` from the branch.